geomstats/geometry/spd_matrices.py

```diff
@@ -848,13 +849,15 @@ def squared_dist(self, point_a, point_b, **kwargs):
         trace_b = gs.trace(point_b)
         trace_prod = gs.trace(sqrt_product)
 
-        return trace_a + trace_b - 2 * trace_prod
+        squared_dist = trace_a + trace_b - 2.0 * trace_prod
+
+        return gs.where(squared_dist < 0.0, 0.0, squared_dist)
```

Comment:
Why would the squared_dist be < 0? 

Are we sure that it is a numerical issue, and not a problem in the math in the code?

If it is the later, then the gs.where would hide that bug, that would otherwise be caught by unit tests.

Reply:
On my tests the negative values were always super close to zero (e.g. -1e-12). the problem seems to be the tolerance in `gs.linalg.sqrtm`.

Your point is very good though!